nexpose/nexpose.py

```diff
@@ -2032,22 +2043,45 @@ def DownloadReport(self, report_or_id, callback_function=None, block_size=DEFAUL
         reader = self.GetReportStreamReader(report_or_id)
         return DownloadFromStreamReader(reader, callback_function, block_size)
 
-    def GenerateScanReport(self, scan_or_id):
+    def GenerateScanReport(self, scan_or_id, format='raw-xml-v2', template_id='audit-report'):
         """
         Generate a report of a scan.
         """
         if isinstance(scan_or_id, ScanSummary):
             scan_or_id = scan_or_id.id
-        data = self.RequestReportAdhocGenerate(scan_or_id)
+        data = self.RequestReportAdhocGenerate(scan_or_id, format, template_id)
         data = self.VerifySuccess(data)
         data = data.tail.replace('\r', '').strip().split('\n')
-        assert data[1] == 'Content-Type: text/xml; name=report.xml'
-        assert data[2] == 'Content-Transfer-Encoding: base64'
-        assert data[3] == ''
-        assert data[0] == data[-1][:-2]
+        boundary_top = data[0]
+        content_type = data[1]
+        encoding = data[2]
         body = ''.join(data[4:-1])
+        boundary_bottom = data[-1]
+        if boundary_top != boundary_bottom[:-2]:
+            raise ValueError("Invalid boundary")
```

Comment:
I replaced the assertions as per Python recommendation: asserts could be compiled out so throwing exceptions is the preferred method here.